src/plugins/console/public/lib/autocomplete/body_completer.ts

```diff
@@ -112,32 +136,45 @@ class ScopeResolver extends SharedComponent {
     return resolvePathToComponents(path, context, editor, components);
   }
 
-  getTerms(context, editor) {
-    const options = [];
+  getTerms(context: AutoCompleteContext, editor: CoreEditor) {
+    const options: Term[] = [];
     const components = this.resolveLinkToComponents(context, editor);
     _.each(components, function (component) {
-      options.push.apply(options, component.getTerms(context, editor));
+      const option = component.getTerms(context, editor);
+      if (option) {
+        options.push.apply(options, option);
+      }
     });
     return options;
   }
 
-  match(token, context, editor) {
-    const result = {
+  match(token: string | string[] | Token, context: AutoCompleteContext, editor: CoreEditor) {
+    const result: MatchResult = {
       next: [],
     };
     const components = this.resolveLinkToComponents(context, editor);
     _.each(components, function (component) {
       const componentResult = component.match(token, context, editor);
       if (componentResult && componentResult.next) {
-        result.next.push.apply(result.next, componentResult.next);
+        result.next?.push.apply(result.next, componentResult.next);
       }
     });
 
     return result;
   }
 }
-function getTemplate(description) {
-  if (description.__template) {
+function getTemplate(description: Description): Description | Template {
+  if (Array.isArray(description)) {
+    if (description.length === 1) {
+      if (_.isObject(description[0])) {
+        // shortcut to save typing
+        const innerTemplate = getTemplate(description[0]);
+
+        return innerTemplate != null ? [innerTemplate] : [];
+      }
+    }
+    return [];
+  } else if (typeof description !== 'string' && description.__template) {
```

Comment:
As I remember, I did this way to preserve completely the same behavior, because of this part
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/adb538ef73b7dd9d5488de6cb46eb7f8a7782661/src/plugins/console/public/lib/autocomplete/body_completer.js#L173-L179
I think we can remove the test and just leave string check because I don't think that `!/^\{.*\}$/.test(description)` does anything, because even if tests fails in the next else statement the description is returned anyway.
I refactored it to reflect that in my latest commit, you can recheck this part to see if everything is good to go.